File: testdata/languages/csharp/Sample.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using System.Text.Json;

namespace SampleNamespace
{
    public enum Status { Pending, Running, Done }

    public interface IRepository<T> where T : class
    {
        Task<T?> FindAsync(string id);
        Task SaveAsync(T entity);
    }

    public record Entity(string Id, string Name)
    {
        public Status Status { get; set; } = Status.Pending;
        public Dictionary<string,string> Meta { get; init; } = new();
    }

    public class Service
    {
        private readonly IRepository<Entity> _repo;
        private readonly Dictionary<string, Entity> _cache = new();

        public Service(IRepository<Entity> repo) => _repo = repo;

        public async Task<Entity> ExecuteAsync(string id)
        {
            if (_cache.TryGetValue(id, out var cached)) return cached;
            var e = await _repo.FindAsync(id) ?? throw new KeyNotFoundException(id);
            e.Status = Status.Running;
            await _repo.SaveAsync(e);
            _cache[id] = e;
            return e;
        }

        public string ToJson(Entity e) => JsonSerializer.Serialize(e);
    }

    public class SampleClass
    {
        public string Name { get; set; } = "";
        public void Execute() => Console.WriteLine($"Hello {Name}");
        public static async Task Main() {
            var svc = new Service(null!);
            Console.WriteLine("ready");
        }
    }
}

File: testdata/languages/cpp/sample.cpp
#include <iostream>
#include <string>
#include <vector>
#include <memory>
#include <unordered_map>
#include <future>
#include <optional>

enum class Status { Pending, Running, Done };

template<typename T>
class Repository {
public:
    virtual ~Repository() = default;
    virtual std::optional<T> find(const std::string& id) = 0;
    virtual void save(const T& e) = 0;
};

struct Entity {
    std::string id;
    std::string name;
    Status status = Status::Pending;
    std::unordered_map<std::string,std::string> meta;
    explicit Entity(std::string i, std::string n): id(std::move(i)), name(std::move(n)) {}
};

class Service {
    std::shared_ptr<Repository<Entity>> repo;
    std::unordered_map<std::string, Entity> cache;
public:
    explicit Service(std::shared_ptr<Repository<Entity>> r): repo(std::move(r)) {}
    std::future<std::optional<Entity>> executeAsync(const std::string& id) {
        return std::async(std::launch::async, [this, id]() -> std::optional<Entity> {
            auto opt = repo->find(id);
            if (!opt) return std::nullopt;
            opt->status = Status::Running;
            repo->save(*opt);
            cache.emplace(id, *opt);
            return opt;
        });
    }
    template<typename U>
    std::vector<U> mapEntities(const std::vector<Entity>& in, std::function<U(const Entity&)> fn) {
        std::vector<U> out; out.reserve(in.size());
        for (auto& e: in) out.push_back(fn(e));
        return out;
    }
};

class Engine {
public:
    void start() { std::cout << "engine start\n"; }
    virtual void stop() {}
    virtual ~Engine() = default;
};

class Worker : public Engine {
    std::vector<std::future<void>> tasks;
public:
    void stop() override { std::cout << "worker stop\n"; }
    void submit(std::function<void()> fn) { tasks.push_back(std::async(fn)); }
};

int main() {
    auto svc = std::make_shared<Service>(nullptr);
    std::cout << "ok\n";
    return 0;
}
